discovery/discovery.py

```diff
@@ -20,39 +23,84 @@
 from utils.file_handler import FileHandler
 from utils.decorators.persist_execution import persistence
 from utils.visualizer import Visualizer
-from utils.metadata import Metadata, ColMetadata
 
 
 class Discovery:
     def __init__(self, discovery_config: dict):
         self.config = discovery_config
         self.file_handler = FileHandler()
-        self.file_metadata = []
+        self.dataframe_file_metadata_pairs = []
         self.visualiser = Visualizer()
 
     def create_visual(self, pathname):
         """
         Build a visual based on stored metadata
         """
-        self.visualiser.draw(self.file_metadata, pathname)
+
+        # TODO: find a fancy python way to do this
+        metadata = []
+        for dataframe, metadatum in self.dataframe_file_metadata_pairs:
+            metadata.append(metadatum)
+        self.visualiser.draw(metadata, pathname)
 
     def reconstruct_metadata(self):
         """
         Builds metadata based on the currently loaded files
         """
         file_metadata = []
-        for path, dataframe in self.file_handler.loaded_files.items():
-            col_meta = []
-            for col_name in dataframe.columns:
-                average, col_min, col_max = None, None, None
-                if is_numeric_dtype(dataframe[col_name]):
-                    average = dataframe[col_name].mean()
-                    col_min = dataframe[col_name].min()
-                    col_max = dataframe[col_name].max()
-                col_meta.append(ColMetadata(col_name, dataframe[col_name].dtype, average, col_min, col_max))
-
-            file_metadata.append(Metadata(path, col_meta))
-        self.file_metadata = file_metadata
+        for path, file_descriptor in self.file_handler.loaded_files.items():
+            metadatum = self._construct_metadatum(file_descriptor)
+            file_metadata.append((file_descriptor["dataframe"], metadatum))
+        self.dataframe_file_metadata_pairs = file_metadata
+
+    def _construct_metadatum(self, file_descriptor):
```

Comment:
metadata construction would probably make more sense in the metadata file itself

Reply:
The metadata class is only a DTO, it should not contain and logic and **please do not put any logic inside it**

Reply:
Metadata can exist as a DTO, but then there should be another script handling metadata, discovery.py should only expose top level interactions. 
Consider this from the point of a documentation standpoint, where someone is reading DiscoveryClient docstrings to figure out what they can do.

Reply:
I will need to think about this for a bit as I have a few thoughts on the issue. 

In a normal scenario I would very much like to keep the DTO clean. However it might not be attainable now. I agree with the idea that there should be something dedicated, that can generate metadata from DFs -> that would essentially be the same behaviour as writing and reading metadata from files (from a design point of view)

My concern is that if we start adding and modifying the metadata class, then these dependant classes won't update automatically.

Best solution, I have right now is to add a bunch of @property decorators to Metadata and go from there - will see.